code/game/machinery/computer/arcade/orion_event.dm

```diff
@@ -0,0 +1,510 @@
+	var/gamerSkill = 0
+	///some other metric that makes it easier to do randoms with skill testing, god really 3 vars guys
+	var/gamerSkillRands = 0
+
+/datum/orion_event/New(_game)
+	. = ..()
+	game = _game
+
+/**
+ * What happens when this event is selected to trigger, sets vars. also can set some event pre-encounter randomization
+ *
+ * Arguments:
+ * * _gamerSkill: gaming skill of the player
+ * * _gamerSkillLevel: gaming skill level of the player
+ */
+/datum/orion_event/proc/on_select(_gamerSkill, _gamerSkillLevel, _gamerSkillRands)
+	SHOULD_CALL_PARENT(TRUE)
+	gamerSkillLevel = _gamerSkillLevel
+	gamerSkill = _gamerSkill
+	gamerSkillRands = _gamerSkillRands
+
+/**
+ * What happens when you respond to this event by choosing one of the buttons
+ *
+ * Arguments:
+ * * choice: name of the button you pressed.
+ */
+/datum/orion_event/proc/response(choice)
+	game.event = null
+
+/**
+ * Some effect that happens to the carbon when this event triggers on an emagged arcade machine.
+ *
+ * By default, it just sends an audible message and a sound, both vars on the orion datum
+ * Arguments:
+ * * gamer: victim to apply effects to
+ * * gamerSkill: skill of the gamer, because gamers gods can lessen the effects of the emagged machine
+ * * gamerSkillLevel: skill level of the gamer, another way to measure emag downside avoidance
+ */
+/datum/orion_event/proc/emag_effect(mob/living/carbon/gamer)
+	if(emag_message)
+		game.audible_message(emag_message)
+	if(emag_sound)
+		playsound(game, emag_sound, 100, TRUE)
+
+///Engine Breakdown - spend one engine part or wait 3 days (harmless emag effect)
+/datum/orion_event/engine_part
+	name = "Engine Breakdown"
+	text = "Oh no! The engine has broken down! \
+	You can repair it with an engine part, or you \
+	can make repairs for 3 days."
+	emag_message = "<span class='warning'>You hear some large object lurch to a halt right behind you! When you go to look, nothing's there...</span>"
+	emag_sound = 'sound/effects/creak1.ogg'
+	weight = 2
+	event_responses = list()
+
+/datum/orion_event/engine_part/on_select(_gamerSkill, _gamerSkillLevel, gamerSkillRands)
+	..()
+	if(game.engine >= 1)
+		event_responses += "Fix Engine"
+	event_responses += "Wait"
+
+/datum/orion_event/engine_part/response(choice)
+	if(choice == "Fix Engine")
+		game.engine = max(0, --game.engine)
+	else
+		game.food -= ((game.alive + game.lings_aboard)*2)*3
+	event_responses.Cut()
+	..()
+
+///Malfunction - spend one engine part or wait 3 days (emag effect randomizes some stats)
+/datum/orion_event/electronic_part
+	name = "Malfunction"
+	text = "The ship's systems are malfunctioning! \
+	You can replace the broken electronics with spares, \
+	or you can spend 3 days troubleshooting the AI."
+	weight = 2
+	//set by select
+	event_responses = list()
+
+/datum/orion_event/electronic_part/on_select(_gamerSkill, _gamerSkillLevel, gamerSkillRands)
+	..()
+	if(game.electronics >= 1)
+		event_responses += "Repair Electronics"
```

Comment:
I don't like the new-then-comparing strings. Can we make these defines as well? Prevents any possibility of typos and such.